### PR TITLE
Comply with coq8.5

### LINKED ISSUE #31: Crash on Coq 8.5
The plugin doesnt work anymore in coq 8.5, is there intention to work on this?

### PR DESCRIPTION
**Pull request by zjhmale**
Fixes #31 and #42.

screenshot

screenshot

now we can kindly play with Coq8.5 :)

@trefis please review.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+# virtualenv
+.venv/
+venv/
+ENV/
+
+# Spyder project settings
+.spyderproject
+
+# Rope project settings
+.ropeproject
```
Comment (trefis) [inline .gitignore]: I don't think you meant to commit this file.
At least not with this content.

File: autoload/coqtop.py [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+            coqtop.terminate()
+            coqtop.communicate()
+        except OSError:
+            pass
+        coqtop = None
+
+def ignore_sigint():
+    signal.signal(signal.SIGINT, signal.SIG_IGN)
+
+def escape(cmd):
```
Comment (trefis) [inline autoload/coqtop.py]: Is that really useful necessary for a vim plugin?

File: autoload/coqtop.py [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+
+    if r.val.val is None:
+        print "no goals"
+        return r
+
+    gs = r.val.val
+    print "goals =>", gs
+    return r
+
+if __name__ == '__main__':
```
Comment (trefis) [inline autoload/coqtop.py]: The following lines look like testing. I don't think you meant to commit them.

File: autoload/coquille.vim [hunk 4]
```diff
@@ -2,7 +2,7 @@ let s:coq_running=0
 let s:current_dir=expand("<sfile>:p:h") 
 
 if !exists('coquille_auto_move')
-    let g:coquille_auto_move="false"
+    let g:coquille_auto_move="true"
```
Comment (trefis) [inline autoload/coquille.vim]: Please don't change things like this. Or at least propose them in another PR so they can be discussed independently.

**Comment by trefis** [comment]
(Sorry for the delay btw)

**Comment by zjhmale** [comment]
@trefis no worries,  i'll take care of your comments first :)
>(Sorry for the delay btw)

File: .gitignore (hunk 1 above)
Comment (zjhmale) [inline .gitignore]: ok, i'll remove this ignore file.

File: autoload/coqtop.py (hunk 2 above)
Comment (zjhmale) [inline autoload/coqtop.py]: i think it is nothing to do with vim. 😅  we can not happily parse xml data using `xml.etree.ElementTree` without escaping these special chars.

image

Comment (zjhmale) [inline autoload/coqtop.py]: @trefis i just move this part to a test file and it works fine on [my fork](https://travis-ci.org/zjhmale/coquille). don't forget to check this project out on travis if you think this approach is acceptable.  😃

**Comment by zjhmale** [comment]
@trefis please review again :)

**Comment by zjhmale** [comment]
poke @trefis

**Comment by pedrotst** [comment]
Omg, you're my hero, thaanks!

File: autoload/coqtop.py [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+              .replace("&#41;", ')')
+
+def get_answer():
+    acc = ''
+    fd = coqtop.stdout.fileno()
+    messageNode = None
+    while True:
+        try:
+            data = os.read(fd, 0x4000)
+            acc += data
```
Comment (zjhmale) [inline autoload/coqtop.py]: Yeah, exactly, I'll remove it, THX!

Comment (gogokefakefa) [inline autoload/coqtop.py]: Isn't it possible that there is a longer input from coqtop (or, for example, the full output is not generated in time) ? I am not familiar with its behavior, but upon executing some local code, the execution of the plugin freezes here.

**Comment by civiclabsconsulting** [comment]
@trefis Bump, please!

**Comment by zjhmale** [comment]
I'll check out if this is also working well on Coq8.6 this weekend.

**Comment by civiclabsconsulting** [comment]
Thanks @zjhmale, that would be fantastic. As a heads up - not sure the patch is exactly working with 8.5. I'm working through a learning text right now (Certified Programs Dependent Types) and it's throwing a different error now with a basic example (`StackMachine.v`). Could be on my end, but not sure. 

After loading a `*.v` file and running `CoqLaunch` then `CoqNext`.

1__tmux__tmux_

1__tmux__tmux_ 2

**Comment by zjhmale** [comment]
@civiclabsconsulting I'll check out that 😄

**Comment by trefis** [comment]
I finally tested this with 8.5 and it "works" (i.e. it seems to work, at least it's much better than not having anything).
I squashed a few commits and removed the commit changing the behavior of automoving the cursor (as it is unrelated, and I don't like it) but otherwise merged as is.

Some commits (like cleaning the "info" window before doing anything) did not seem so great to me, but I kept them anyway.

Thanks a lot for the contribution!

**Comment by trefis** [comment]
Btw, I just encountered the same kind of bug as @civiclabsconsulting (i.e. errors not being displayed properly, but a python backtrace being displayed in vim).

**Pull request closed without merging**